services/ext/service.go

```diff
@@ -556,46 +560,223 @@ func (s *Service) CanProvideCollectibleMetadata(id thirdparty.CollectibleUniqueI
 	}
 
-	if community != nil {
-		tokensMetadata := community.CommunityTokensMetadata()
-
-		for _, tokenMetadata := range tokensMetadata {
-			contractAddresses := tokenMetadata.GetContractAddresses()
-			if contractAddresses[uint64(id.ContractID.ChainID)] == id.ContractID.Address.Hex() {
-				return &thirdparty.FullCollectibleData{
-					CollectibleData: thirdparty.CollectibleData{
-						ID:          id,
-						Name:        tokenMetadata.GetName(),
-						Description: tokenMetadata.GetDescription(),
-						ImageURL:    tokenMetadata.GetImage(),
-						TokenURI:    tokenURI,
-					},
-					CollectionData: &thirdparty.CollectionData{
-						ID:       id.ContractID,
-						Name:     tokenMetadata.GetName(),
-						ImageURL: tokenMetadata.GetImage(),
-					},
-				}, nil
+	return community, nil
+}
+
+func (s *Service) fetchCommunityToken(communityID string, contractID thirdparty.ContractID) (*token.CommunityToken, error) {
+	if s.messenger == nil {
+		return nil, fmt.Errorf("messenger not ready")
+	}
+
+	return s.messenger.GetCommunityToken(communityID, int(contractID.ChainID), contractID.Address.String())
+}
+
+func (s *Service) fetchCommunityCollectibleMetadata(community *communities.Community, contractID thirdparty.ContractID) (*protobuf.CommunityTokenMetadata, error) {
+	tokensMetadata := community.CommunityTokensMetadata()
+
+	for _, tokenMetadata := range tokensMetadata {
+		contractAddresses := tokenMetadata.GetContractAddresses()
+		if contractAddresses[uint64(contractID.ChainID)] == contractID.Address.Hex() {
+			return tokenMetadata, nil
+		}
+	}
+
+	return nil, nil
+}
+
+func (s *Service) fetchCommunityCollectiblePermission(community *communities.Community, id thirdparty.CollectibleUniqueID) (*communities.CommunityTokenPermission, error) {
+	permissionTypes := []protobuf.CommunityTokenPermission_Type{
+		protobuf.CommunityTokenPermission_BECOME_TOKEN_OWNER,
+		protobuf.CommunityTokenPermission_BECOME_TOKEN_MASTER,
+	}
+
+	// Check permission types of interest
+	for _, permissionType := range permissionTypes {
+		// See if any community permission matches the type we're looking for
+		for _, permission := range community.TokenPermissionsByType(permissionType) {
+			// See if any token criterion matches the collectible we're looking for
+			for _, tokenCriterion := range permission.TokenCriteria {
+				if tokenCriterion.Type == protobuf.CommunityTokenType_ERC721 {
+					for chainID, contractAddressStr := range tokenCriterion.ContractAddresses {
+						contractAddress := commongethtypes.HexToAddress(contractAddressStr)
+						if chainID == uint64(id.ContractID.ChainID) && contractAddress == id.ContractID.Address {
+							for _, tokenID := range tokenCriterion.TokenIds {
+								tokenIDBigInt := new(big.Int).SetUint64(tokenID)
+								if id.TokenID.Cmp(tokenIDBigInt) == 0 {
+									return permission, nil
+								}
+							}
+						}
+					}
+				}
 			}
 		}
 	}
 
 	return nil, nil
 }
+
+func boolToString(value bool) string {
```

Comment:
Traits are generic string pairs (Title + Value) that do not follow any kind of standard. We don't do any parsing on the client side, we just display them as they are. If we want this to look like in Figma, we need to set it in its text form.

Reply:
Hmm we don't translate other collectible traits, not sure if we want to handle these in a special way. I'll ask Ben!

Reply:
So... My logic here was to try to make this work as much as possible as a standard collectible and not require a bunch of special logic on the client side. 
These traits are just cosmetic elements, we're not supposed to use them for any business logic. If we needed to use the data for some functionality, I'd transfer that in some different struct.
Moreover, we're not translating these strings just like we don't translate types/values for other collectibles. I discussed this with Ben and he said:
```
It feels practical at least for MVP that we opt to keep it in English just as Opensea does, even if the user changes their language setting. I think the fact NFT communities often have their own quirky terminology and language for traits makes it more ok for us to avoid devling into translation - examples attached - eg skelly cig, puffer up
```